samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java

```diff
@@ -27,82 +30,105 @@
-	private IFCMNotificationService service = NotificationFactory
+	private String accountKey = null;
+	private String msgAuthKey = null;
+	private ISmsProvider smsProvider = null;
+	private IFCMNotificationService ifcmNotificationService = NotificationFactory
 			.getInstance(NotificationFactory.instanceType.httpClinet.name());
 	private static final String FCM_ACCOUNT_KEY = "fcm_account_key";
+	private static final String MSG_AUTH_KEY = "msg_auth_key";
+
 	public void initialize(Config config) throws Exception {
 		JSONUtils.loadProperties(config);
 		appConfig = config;
 		Logger.info("NotificationService:initialize: Service config initialized");
+		accountKey = appConfig.get(FCM_ACCOUNT_KEY);
+		msgAuthKey = appConfig.get(MSG_AUTH_KEY);
+		SMSConfig smsConfig = new SMSConfig(msgAuthKey, "");
+		smsProvider = SMSFactory.getInstance("91SMS", smsConfig);
 	}
 
 	public void processMessage(Map<String, Object> message) throws Exception {
-		String accountKey = appConfig.get(FCM_ACCOUNT_KEY);
 		Logger.info("Account key:"+ accountKey);
 		FCMHttpNotificationServiceImpl.setAccountKey(accountKey);
 		Map<String, String> notificationMap = new HashMap<String, String>();
 		Map<String, Object> edataMap = (Map<String, Object>) message.get(Constant.EDATA);
 		String requestHash = "";
-		List<String> deviceIds = new ArrayList<String>();
+		List<String> deviceIds = null;
 		String topic = null;
+		boolean isSuccess = false;
 		if (edataMap != null && edataMap.size() > 0) {
 			String actionValue = (String) edataMap.get(Constant.ACTION);
 			if (Constant.ACTION_NAME.equalsIgnoreCase(actionValue)) {
 				Map<String, Object> requestMap = (Map<String, Object>) edataMap.get(Constant.REQUEST);
 				requestHash = OneWayHashing.encryptVal(mapper.writeValueAsString(requestMap));
-				Map<String, Object> tmp = (Map<String, Object>) requestMap.get(Constant.NOTIFICATION);
-				if (tmp.get(Constant.IDS) != null) {
-					deviceIds = (List<String>) tmp.get(Constant.IDS);
+				Map<String, Object> objectMap = (Map<String, Object>) message.get(Constant.OBJECT);
+				if (!requestHash.equals((String) objectMap.get(Constant.ID))) {
+					Logger.info("NotificationService:processMessage: hashValue is not matching - " + requestHash);
 				} else {
-					Map<String, Object> configMap = (Map<String, Object>) tmp.get(Constant.CONFIG);
-					topic = (String) configMap.getOrDefault(Constant.TOPIC, "");
+					Map<String, Object> tmp = (Map<String, Object>) requestMap.get(Constant.NOTIFICATION);
+					if (tmp.get(Constant.IDS) != null) {
+						deviceIds = (List<String>) tmp.get(Constant.IDS);
+					}
+					if(tmp.get(Constant.MODE).equals("phone")) {
+						if(deviceIds != null) {
+							Map<String, Object> templateMap = (Map<String, Object>) tmp.get(Constant.TEMPLATE);
+							String smsData = (String) templateMap.get(Constant.DATA);
+							isSuccess = sendSmsNotification(deviceIds, smsData);
+						} else {
+							Logger.info("mobile numbers not provided");
```

Comment:
if ids are not provided, shouldn't there be an exception?

Reply:
our perspective is message should be checked before sending it to kafka, so we just logging failure message